### PR TITLE
Closes #1875 duration and interval time difference types update

### LINKED ISSUE #1875: Feature Request:  Enhance `compute_duration()` to handle Interval and Duration
### Feature Idea

`lubridate::time_length(x, unit = "second")` handles [Interval](https://lubridate.tidyverse.org/reference/Interval-class.html) and [Duration](https://lubridate.tidyverse.org/reference/Duration-class.html) differently when unit is months/years

> When x is an [Interval](https://lubridate.tidyverse.org/reference/Interval-class.html) object and unit are years or months, time_length() takes into account the fact that all months and years don't have the same number of days.
> When x is a [Duration](https://lubridate.tidyverse.org/reference/Duration-class.html), [Period](https://lubridate.tidyverse.org/reference/Period-class.html) or [difftime()](https://rdrr.io/r/base/difftime.html) object, length in months or years is based on their most common lengths in seconds (see [timespan()](https://lubridate.tidyverse.org/reference/timespan.html)).

Example of difference between intervals and durations:
```r
> int <- interval(ymd("2000-02-01"), ymd("2000-03-01"))
> time_length(int, "months")
[1] 1
> time_length(as.duration(int), "months")
[1] 0.9527721
```

Currently, `compute_duration()` only uses interval to calculate "duration", would it be okay to add a feature to choose between interval and duration?

https://github.com/pharmaverse/admiral/blob/6dead5ad9b9fe529d8b459ae718f8f35cb52296f/R/compute_duration.R#L158

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by ddsjoberg**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #1875 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/compute_duration.R [hunk 1]
```diff
@@ -66,10 +66,38 @@
 #'
 #'   Permitted Values: `TRUE`, `FALSE`
 #'
+#' @param type lubridate duration type.
+#'
+#'   See below for details.
+#'
+#'   Default: `"duration"`
```
Comment (zdz2101) [inline R/compute_duration.R:73]:
```suggestion
```

No longer including Default in roxygen documentation

File: R/compute_duration.R [hunk 2]
```diff
@@ -66,10 +66,38 @@
...
+#'
+#'   Permitted Values: `"duration"`, `"interval"`
```
Comment (zdz2101) [inline R/compute_duration.R:75]:
```suggestion
#'   _Permitted Values_: `"duration"`, `"interval"`
```

**Review by zdz2101 — COMMENTED** [review]
Super small stuff, otherwise I think the pharmaverse sdtm is now going through

**Review by zdz2101 — APPROVED** [review]

**Comment by ddsjoberg** [comment]
I renamed the branch, but GH didn't pick up the new name. I am going to close this PR and re-open with the newly named branch.

**Pull request closed without merging**